backend/src/apiserver/storage/run_store.go

```diff
@@ -365,19 +365,26 @@ func (s *RunStore) UpdateRun(runID string, condition string, workflowRuntimeMani
 	// new in the status of an Argo manifest. This means we need to keep track
 	// manually here on what the previously updated state of the run is, to ensure
 	// we do not add duplicate metadata. Hence the locking below.
-	row := tx.QueryRow("SELECT WorkflowRuntimeManifest FROM run_details WHERE UUID = ? FOR UPDATE", runID)
+	var query string
```

Comment:
Could we have an interface that abstracts away the differences between the MySQL and SQLLite DB? I think it would look cleaner than having switch statements.

Reply:
The difference in behaviour is real and not abstract, and in the interest of code readability, it's worth making this explicit. This makes it clear to the reader that with mysql, we do a row lock, while no such locking takes place with sqlite. That's a fundamental difference that I think is worth highlighting to the reader, rather than hide it behind some abstraction. It's crystal clear here that row locking path is never tested in unit tests as well.

Reply:
I moved this to the db interface, FWIW. PTAL.